gateway/bucketfs.go

```diff
@@ -47,22 +49,38 @@ func serveBucket(fs serveBucketFileSystem) gin.HandlerFunc {
 			} else {
 				c.Abort()
 			}
+		} else if target != "" {
+			c.Writer.WriteHeader(http.StatusOK)
+			c.Writer.Header().Set("Content-Type", "text/html")
+			if err := fs.Write(bucket, fmt.Sprintf("%s/%s", c.Request.URL.Path, target), c.Writer); err != nil {
+				abort(c, http.StatusInternalServerError, err)
+			} else {
+				c.Abort()
+			}
 		}
+
 	}
 }
 
-func (f *bucketFileSystem) Exists(bucket, pth string) bool {
+func (f *bucketFileSystem) Exists(bucket, pth string) (bool, string) {
 	if bucket == "" || pth == "/" {
-		return false
+		return false, ""
 	}
 
 	ctx, cancel := context.WithTimeout(context.Background(), f.timeout)
 	defer cancel()
 	rep, err := f.client.ListBucketPath(ctx, "", path.Join(bucket, pth), f.auth)
 	if err == nil && !rep.Item.IsDir {
-		return true
+		return true, ""
+	}
+	if rep.Item.IsDir {
+		for _, item := range rep.Item.Items {
+			if item.Name == "index.html" {
+				return false, item.Name
```

Comment:
this could return `(bool, bool)` but i left it open here intentionally to support different default render paths. at least `.htm` and `.html` came to mind

Reply:
To clarify, the idea here is to allow rendering a website in a bucket directory, yes? Are relative URLs broken when you do this?

Reply:
i tested by recreating this bucket's data locally: https://gatsby-ipfs-blog.textile.cafe/

navigating links from the homepage worked fine, all embedded assets worked fine (imgs, js, etc)

also, direct links to subdirectories (the issue on master) was fixed. so linking to https://gatsby-ipfs-blog.textile.cafe/new-beginnings/ works on this branch (not on production)

Reply:
Ahh, I see... I had forgotten the core issue. Sweet.